### PR TITLE
Chore/Build and Test GitHub action

### PR DESCRIPTION
**Pull request by GregBal**
Adds a new github action that builds and tests the library.
This action is set to run on main branch merges, and as a check on PRs

### CODE REVIEW

**Review by chris-asl — COMMENTED** [review]
Looks good Greg! 👏 

I've left some minor comments to check

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17.0.1
+          distribution: 'adopt'
```
Comment (kiraleos) [inline .github/workflows/ci.yml:22]: it probably doesn't matter but shouldn't we use the same distribution we use for atlas (zulu)?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: CI
+concurrency:
+  group: ${{ github.head_ref }}
```
[the remark this answers is not in the document]
Comment (GregBal) [reply .github/workflows/ci.yml]: Yes the intention of this is to cancel concurrent runs for each pr,
I fixed it the way the docs suggest

**Comment by ippokratoys** [comment]
Nice Greg! 👏🏾 

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

We could also add badges (GitHub docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge)).

It's would be a great idea to find an open-source project (ideally a library), see what they have (filter out what would be useful) and add a ticket for them or add it (some of them are easy wins).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17.0.1
+          distribution: 'adopt'
+      - name: Build and test
```
[the remark this answers is not in the document]
Comment (chris-asl) [reply .github/workflows/ci.yml:23]: Greg, can we somehow see the reports before merging to main?

[the remark this answers is not in the document]
Comment (GregBal) [reply .github/workflows/ci.yml:23]: Yes you can view them by clicking on the PR check details -> Summary -> there are 2 uploaded zip files.
Just show the action you sent me, will try if it's more convenient

**Review by chris-asl — APPROVED** [review]
LGTM mate! Great job! 👏

**Pull request merged**